csrc/multidevice/executor.h

```diff
@@ -139,6 +135,10 @@ class MultiDeviceExecutor {
   std::unordered_map<SegmentedGroup*, bool> should_run_;
   // Cache whether a SegmentedGroup requires inter-device communication
   std::unordered_map<SegmentedGroup*, bool> is_resharding_;
+  // Cached objects used for MultiDevice allocation
+  std::unique_ptr<kir::Kernel> allocator_kernel_;
+  std::vector<Val*> vals_to_allocate_;
```

Comment:
Can you add a code comment for `vals_to_allocate`? I didn't review the initial PR where you created it, so I'm lost on its meaning.